packages/code-du-travail-frontend/src/outils/IndemniteLicenciement/index.js

```diff
@@ -1,60 +1,51 @@
-import React, { useReducer, useState } from "react";
+import React from "react";
 import PropTypes from "prop-types";
 import { OnChange } from "react-final-form-listeners";
 import { Section, Container, Wrapper } from "@cdt/ui";
 
-import { StepReducer, getInitialSteps } from "./reducer";
+import { stepReducer, initialSteps } from "./stepReducer";
 import { Wizard } from "./Wizard";
 
 function CalculateurIndemnite() {
-  const initialSteps = getInitialSteps();
-  const [steps, dispatch] = useReducer(StepReducer, initialSteps);
-  const [values, setValues] = useState({});
   /**
-   * The rules defined here allows to manage additionnal steps to the form
+   * The rules defined here allows to manage additionnal steps to the wizard
    */
-  const rules = [
-    <OnChange key="rule-same-salaire" name="hasSameSalaire">
-      {value =>
-        value === false
-          ? dispatch({ type: "add_primes" })
-          : dispatch({ type: "remove_primes" })
-      }
-    </OnChange>,
-    <OnChange key="rule-branche" name="branche">
-      {async value => {
-        if (value) {
-          const module = await import(`./ccn/${value}`);
-          const steps = module.steps.filter(({ condition = () => true }) =>
-            condition(values)
-          );
-          dispatch({ type: "add_branche", payload: steps });
-        } else {
-          dispatch({ type: "remove_branche" });
+  const Rules = ({ values, dispatch }) => (
```

Comment:
to make sure we have access to values and dispatch which are in the wizard below, I had to transform `rules` into a component